posts/creating-your-first-truffle-project.md

```diff
@@ -0,0 +1,634 @@
+        
+        
+        contract('Migrations', (accounts) => {
+        
+            let migrations;
+        
+            // build up and tear down a new Migrations before each test
+            beforeEach(async () => {
+                migrations = await Migrations.deployed();
+                console.log(migrations);
+            });
+        
+            it('has a validated contract size', async () => {
+                // bytecode is in hexadecimal, where each byte is represented by two characters: 0x00 -> 0xff
+                let bytecodeSize = migrations.constructor._json.bytecode.length / 2;
+                let deployedBytecodeSize = migrations.constructor._json.deployedBytecode.length / 2;
+        
+                console.info('Migrations deployed at address: ' + web3.utils.toChecksumAddress(migrations.address))
+                console.info(' -- size of bytecode in bytes = ', bytecodeSize);
+                console.info(' -- size of deployed in bytes = ', deployedBytecodeSize);
+                console.info(' -- initialisation and constructor code in bytes = ', bytecodeSize - deployedBytecodeSize);
+        
+                // Make assertion on deployed since the initial transaction takes constructor bytecode into account
+                assert(deployedBytecodeSize <= MAX_DEPLOYED_BYTECODE_SIZE, 'Contract bytecode is too big to deploy!');
+            });
+        
+        });
+    Here is an acceptance test. It can be writen for every contract because it makes no assumptions about the contract functions or data. Instead, it tests the compiled bytecode's size. It is important when writing smart contracts, to be aware of the transactions with the blockchain because not only do you pay for every byte of code, but some transactions are too large and will fail.
+
+5. We can also write a unit test
+
+        it('sets lastCompletedMigration by the owner', async () => {
+            let expectedCompleted = 1234;
+    
+            await migrations.setCompleted(expectedCompleted, { from: accounts[0] });
+    
+            assert.equal(expectedCompleted, await migrations.last_completed_migration({ from: accounts[0] }),
+                'setComplete did not update last_completed_migration');
+        });
+    Here is a unit test that tests the happy path for the function `setCompleted`. Notice the use of async and await in these tests. Every time we call the contract, we must wait for our client to respond. 
+
+6. Run the tests
+
+        npm run test
+    ![Truffle test (missing pic)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/truffle_test_1.png)
+    
+    Here, we don't see any listed methods in the table. This is because the only non-contract-creation transactions we tested was `last_completed_migration` which was a [*view*](https://solidity.readthedocs.io/en/v0.5.2/contracts.html#view-functions) function, which does not get picked up by Mocha.
```

Comment:
Grammar: change
the only non-contract-creation transactions we tested was `last_completed_migration` which was a [*view*](https://solidity.readthedocs.io/en/v0.5.2/contracts.html#view-functions) function, which does not get picked up by Mocha.
 to
the only non-contract-creation transaction we tested was `last_completed_migration`, which is a [*view*](https://solidity.readthedocs.io/en/v0.5.2/contracts.html#view-functions) function that does not get picked up by Mocha.